src/Evolution/DgSubcell/PerssonTci.cpp

```diff
@@ -23,27 +23,61 @@ namespace evolution::dg::subcell::detail {
 template <size_t Dim>
 bool persson_tci_impl(const gsl::not_null<DataVector*> filtered_component,
                       const DataVector& component, const Mesh<Dim>& dg_mesh,
-                      const double alpha, const double zero_cutoff) {
+                      const double alpha) {
   ASSERT(component.size() == dg_mesh.number_of_grid_points(),
          "The tensor components being checked must have the same number of "
          "grid points as the DG mesh. The tensor has "
              << component.size() << " grid points while the DG mesh has "
              << dg_mesh.number_of_grid_points() << " grid points.");
 
+  const double component_norm{l2Norm(component)};
+
   const Matrix identity{};
   for (size_t d = 0; d < Dim; ++d) {
     auto matrices = make_array<Dim>(std::cref(identity));
     gsl::at(matrices, d) = Spectral::filtering::zero_lowest_modes(
         dg_mesh.slice_through(d), dg_mesh.extents(d) - 1);
     apply_matrices(filtered_component, matrices, component, dg_mesh.extents());
 
-    // Avoid taking logs of small numbers (or worse, zero)
-    if (l2Norm(*filtered_component) <= zero_cutoff * l2Norm(component)) {
-      continue;
-    }
-
-    if (std::log10(l2Norm(*filtered_component) / l2Norm(component)) >
-        -alpha * std::log10(static_cast<double>(dg_mesh.extents(d) - 1))) {
+    //
+    // Fix note by Yoonsoo Kim, Oct 2021 :
```

Comment:
Why `Fix note`? I don't think this "fixes" anything. The algorithm worked before. How about just `Note:`?

The original algorithm was geared for fluid dynamics where you never get zero density because the physics equations aren't valid there :)